flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1278,7 +1278,11 @@ public void run() {
     }
 
     void scheduleTriggerRequest() {
-        timer.execute(this::executeQueuedRequest);
+        if (isShutdown()) {
+            LOG.debug("Skip scheduling trigger request because is shutting down");
+        } else {
+            timer.execute(this::executeQueuedRequest);
+        }
     }
```

Comment:
I think we are missing a test for this behaviour here.

Reply:
Added a test. I initially assumed this change would be only a hotfix.